gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -2174,57 +2166,146 @@ unsigned char Movable::GetNextFace()
 	return Orientation;
 }
 
-// returns whether we made all pending steps (so, false if we must be called again this tick)
-// we can't just do them all here because the caller might have to update searchmap etc
-bool Movable::DoStep(unsigned int walk_speed, ieDword time)
+
+void Movable::Backoff()
+{
+	randomBackoff = RAND(MAX_PATH_TRIES, MAX_PATH_TRIES * 2);
+}
+
+
+void Movable::BumpAway()
 {
+	area->ClearSearchMapFor(this);
+	area->AdjustPositionNavmap(Pos);
+}
+
+// Takes care of movement and actor bumping, i.e. gently pushing blocking actors out of the way
+// Returns whether the actor made all pending steps (so, false if it must be called again this tick).
+// The movement logic is a proportional regulator: the displacement/movement vector has a
+// fixed radius, based on actor walk speed, and its direction heads towards the next waypoint.
+// The bumping logic checks if there would be a collision if the actor was to move according to this
+// displacement vector and then, if that is the case, checks if that actor can be bumped
+// In that case, it bumps it and goes on with its step, otherwise it either stops and waits
+// for a random time (inspired by network media access control algorithms) or just stops if
+// the goal is close enough.
+bool Movable::DoStep(unsigned int walkScale, ieDword time) {
+	// Magical values from trial-and-error in order to match the speeds from the original game as closely as possible
+	// See https://github.com/gemrb/gemrb/issues/106#issuecomment-475985677
+	int stepTime = 566;
```

Comment:
I don't know how we came up with this number and it doesn't register in chat. :s It's even odd that it's even. Before we had something like 1500/walkScale, where the scale is usually 9 and now this replaced it, which is fine. But it's clearly not a derivation. Do you have any memory/log of this?

Reply:
"I found two values that seem to give good results for both pst and bg1, I'll check with pst, bg1 and bg2 properly (like 10 tries and give you the data)", from our chat on 2019-03-26. What I did is literally trial-and-error, running [this script here](https://github.com/gemrb/gemrb/issues/106#issuecomment-475985677) using GemRB with my pathfinder, tweaking that value and seeing if the speeds matched.

Reply:
ah, ok.